<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chih-Hsuan Wu" />


<title>Data and Clusters</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEanalysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/C-HW/DEanalysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Data and Clusters</h1>
<h4 class="author">Chih-Hsuan Wu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>DEanalysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230508code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230508)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230508code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230508)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Google Drive/My Drive/spatial/10X/DEanalysis
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCHWDEanalysistree366cd53204c6da534d7b31df1350afc7cc2f8507targetblank366cd53a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/C-HW/DEanalysis/tree/366cd53204c6da534d7b31df1350afc7cc2f8507" target="_blank">366cd53</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomCHWDEanalysistree366cd53204c6da534d7b31df1350afc7cc2f8507targetblank366cd53a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/C-HW/DEanalysis/tree/366cd53204c6da534d7b31df1350afc7cc2f8507" target="_blank">366cd53</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/.Rhistory
    Untracked:  analysis/data_clusters.Rmd
    Untracked:  analysis/group12_13.Rmd
    Untracked:  analysis/group2_19.Rmd
    Untracked:  analysis/group8_17&2_19.Rmd
    Untracked:  analysis/methods_details.Rmd
    Untracked:  analysis/new_criteria.Rmd
    Untracked:  data/10X_inputdata.RData
    Untracked:  data/10X_inputdata_DEresult.RData
    Untracked:  data/10X_inputdata_cpm.RData
    Untracked:  data/10X_inputdata_integrated.RData
    Untracked:  data/10X_inputdata_lognorm.RData
    Untracked:  data/10Xdata_annotate.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/data_clusters.Rmd</code>) and HTML (<code>docs/data_clusters.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/366cd53204c6da534d7b31df1350afc7cc2f8507/docs/data_clusters.html" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/13d726dde0214c2d74705c21397e8bb2f49f8b6f/docs/data_clusters.html" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add DE results on different groups
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/fc9f4b6a78f05402b94483374fba17363245b131/docs/data_clusters.html" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add new_criteria
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/75869536c17b66292074ac0e2e25deaf985d149e/docs/data_clusters.html" target="_blank">7586953</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
<td>
add data_clusters
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-introduction" class="section level1">
<h1>Data introduction</h1>
<p>In this project, we consider a scRNA-seq dataset containing human NK1, NK2 and NK3 T cells. There are a total of 4553 cells contributed by 5 donors, and 29382 genes sequenced. The raw counts are UMI counts generated from 10X protocols.</p>
<p>There are three different datasets used as inputs in this project.</p>
<ul>
<li>Raw data: The raw UMI counts generated from 10X protocols<br />
</li>
<li>Seurat normalized data: The normalized counts from each input sample via ‘Seurat::NormalizeData(x, normalization.method = “LogNormalize”, scale.factor = 10000)’<br />
</li>
<li>Integrated data: The normalized data removing batch effects (only containing 2000 genes)<br />
</li>
<li>CPM data: CPM (Counts Per Million) are obtained by dividing counts by the library counts sum and multiplying the results by a million.</li>
</ul>
</div>
<div id="library-size" class="section level1">
<h1>Library size</h1>
<pre class="r"><code>setwd(&quot;~/Google Drive/My Drive/spatial/10X/DEanalysis&quot;)
alldata = readRDS(&quot;./data/10Xdata_annotate.rds&quot;)
AllCelltypePrep = subset(alldata, subset = expCond2 %in% c(&quot;D3&quot;,&quot;D4&quot;,&quot;D5&quot;,&quot;D6&quot;,&quot;D7&quot;))
rm(alldata)
gc()</code></pre>
<pre><code>             used   (Mb) gc trigger    (Mb) limit (Mb)   max used    (Mb)
Ncells    7396517  395.1   11716810   625.8         NA   11429539   610.5
Vcells 1154081066 8805.0 1887062927 14397.2      16384 1887061169 14397.2</code></pre>
<pre class="r"><code>AllCelltypePrep@active.ident = recode(AllCelltypePrep@active.ident,
                                      &quot;ST1&quot; = &quot;ST&quot;,
                                      &quot;ST2&quot; = &quot;ST&quot;,
                                      &quot;ST3&quot; = &quot;ST&quot;,
                                      &quot;ST4&quot; = &quot;ST&quot;,
                                      &quot;ST5&quot; = &quot;ST&quot;,
                                      &quot;P/V1&quot; = &quot;P/V&quot;,
                                      &quot;P/V2&quot; = &quot;P/V&quot;,
                                      &quot;P/V3&quot; = &quot;P/V&quot;,
                                      &quot;EN1&quot; = &quot;EN&quot;,
                                      &quot;EN2&quot; = &quot;EN&quot;,
                                      &quot;EN3&quot; = &quot;EN&quot;,
                                      &quot;EN4&quot; = &quot;EN&quot;,
                                      &quot;T/NK1&quot; = &quot;T/NK&quot;,
                                      &quot;T/NK2&quot; = &quot;T/NK&quot;,
                                      &quot;T/NK3&quot; = &quot;T/NK&quot;,
                                      )

df = data.frame(libsize = colSums(AllCelltypePrep@assays$RNA@counts), celltype = AllCelltypePrep@active.ident, donor = AllCelltypePrep@meta.data$expCond2)

ggplot(df, aes(x = factor(celltype,levels = levels(AllCelltypePrep@active.ident)), y=libsize, fill = factor(donor, levels = c(&quot;D7&quot;, &quot;D6&quot;, &quot;D5&quot;, &quot;D4&quot;, &quot;D3&quot;)))) + geom_violin() + theme_minimal() + ylab(&quot;library size&quot;) + xlab(&quot;&quot;) +  scale_fill_discrete(name = &quot;donors&quot;, labels=c(&quot;D7&quot;, &quot;D6&quot;, &quot;D5&quot;, &quot;D4&quot;, &quot;D3&quot;)) + theme_bw() + theme(panel.grid = element_blank())</code></pre>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/f3137642695f910c96d11d8c57e0d880826d853a/docs/figure/data_clusters.Rmd/unnamed-chunk-3-1.png" target="_blank">f313764</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-expression-summary" class="section level1">
<h1>Gene expression summary</h1>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-4-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/f3137642695f910c96d11d8c57e0d880826d853a/docs/figure/data_clusters.Rmd/unnamed-chunk-4-1.png" target="_blank">f313764</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>zoom in raw data <img src="figure/data_clusters.Rmd/unnamed-chunk-5-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/f3137642695f910c96d11d8c57e0d880826d853a/docs/figure/data_clusters.Rmd/unnamed-chunk-5-1.png" target="_blank">f313764</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There’s one highly expressed gene <em>MALAT1</em> in raw data. After normalization, the range of the counts changes a lot. (The integrated data doesn’t contain <em>MALAT1</em>) <img src="figure/data_clusters.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="hippo-cluster-result" class="section level1">
<h1>Hippo cluster result</h1>
<p>We applied HIPPO (Heterogeneity-Inspired Pre-Processing tOol) on the raw counts to get 20 clusters. Especially, cluster 2, 8, 12, 13, 17, 19 will be used to demonstrate our poisson glmm DE methods.</p>
<div id="umap" class="section level2">
<h2>UMAP</h2>
<div id="hippo-procedure" class="section level3">
<h3>Hippo procedure</h3>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/f3137642695f910c96d11d8c57e0d880826d853a/docs/figure/data_clusters.Rmd/unnamed-chunk-11-1.png" target="_blank">f313764</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="interested-groups" class="section level3">
<h3>Interested groups</h3>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-12-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/13d726dde0214c2d74705c21397e8bb2f49f8b6f/docs/figure/data_clusters.Rmd/unnamed-chunk-12-1.png" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/f3137642695f910c96d11d8c57e0d880826d853a/docs/figure/data_clusters.Rmd/unnamed-chunk-12-1.png" target="_blank">f313764</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="zero-proportion-plot" class="section level2">
<h2>Zero proportion plot</h2>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="donor-distribution" class="section level2">
<h2>Donor distribution</h2>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/fc9f4b6a78f05402b94483374fba17363245b131/docs/figure/data_clusters.Rmd/unnamed-chunk-14-1.png" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="donor-effect-variation" class="section level1">
<h1>Donor effect variation</h1>
<p>To illustrate the differences in contribution of variation across different datasets, we employed linear regression models (lm(log2(counts + 1) ~ donor + celltype)) to analyze cells in group 12 and group 13. The donor variation and celltype variation were determined by calculating the variances of their respective components. Additionally, the res variation was obtained by squaring the residual standard error. The following plots exhibit the top 500 genes with the highest donor variations, showcasing the contributions of these variations as percentages. The genes were organized into bins based on the quantiles of donor variations. Within each bin, the representative percentage was determined by calculating the median value.</p>
<p>The integration of data did partially reduce the donor variations, although it did not eliminate them completely. However, it is worth noting that the normalization and batch effect removal processes employed also resulted in a reduction in celltype variation. This reduction in celltype variation may pose challenges for conducting further differential expression (DE) analysis.</p>
<p><img src="figure/data_clusters.Rmd/unnamed-chunk-17-1.png" width="1344" style="display: block; margin: auto;" /><img src="figure/data_clusters.Rmd/unnamed-chunk-17-2.png" width="1344" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] SeuratObject_4.1.3          Seurat_4.3.0               
 [3] reshape_0.8.9               SingleCellExperiment_1.20.1
 [5] SummarizedExperiment_1.28.0 Biobase_2.58.0             
 [7] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
 [9] IRanges_2.32.0              S4Vectors_0.36.2           
[11] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[13] matrixStats_0.63.0          ggpubr_0.6.0               
[15] dplyr_1.1.2                 ggplot2_3.4.2              

loaded via a namespace (and not attached):
  [1] backports_1.4.1        workflowr_1.7.0        plyr_1.8.8            
  [4] igraph_1.4.2           lazyeval_0.2.2         sp_1.6-0              
  [7] splines_4.2.2          listenv_0.9.0          scattermore_0.8       
 [10] digest_0.6.31          htmltools_0.5.5        fansi_1.0.4           
 [13] magrittr_2.0.3         tensor_1.5             cluster_2.1.4         
 [16] ROCR_1.0-11            limma_3.54.2           globals_0.16.2        
 [19] spatstat.sparse_3.0-1  colorspace_2.1-0       ggrepel_0.9.3         
 [22] xfun_0.39              RCurl_1.98-1.12        jsonlite_1.8.4        
 [25] progressr_0.13.0       spatstat.data_3.0-1    survival_3.5-5        
 [28] zoo_1.8-12             glue_1.6.2             polyclip_1.10-4       
 [31] gtable_0.3.3           zlibbioc_1.44.0        XVector_0.38.0        
 [34] leiden_0.4.3           DelayedArray_0.24.0    car_3.1-2             
 [37] future.apply_1.10.0    abind_1.4-5            scales_1.2.1          
 [40] edgeR_3.40.2           DBI_1.1.3              spatstat.random_3.1-4 
 [43] rstatix_0.7.2          miniUI_0.1.1.1         Rcpp_1.0.10           
 [46] viridisLite_0.4.2      xtable_1.8-4           reticulate_1.28       
 [49] htmlwidgets_1.6.2      httr_1.4.5             RColorBrewer_1.1-3    
 [52] ellipsis_0.3.2         ica_1.0-3              farver_2.1.1          
 [55] pkgconfig_2.0.3        uwot_0.1.14            deldir_1.0-6          
 [58] sass_0.4.5             locfit_1.5-9.7         utf8_1.2.3            
 [61] labeling_0.4.2         tidyselect_1.2.0       rlang_1.1.1           
 [64] reshape2_1.4.4         later_1.3.0            munsell_0.5.0         
 [67] tools_4.2.2            cachem_1.0.8           cli_3.6.1             
 [70] generics_0.1.3         broom_1.0.4            ggridges_0.5.4        
 [73] evaluate_0.20          stringr_1.5.0          fastmap_1.1.1         
 [76] goftest_1.2-3          yaml_2.3.7             knitr_1.42            
 [79] fs_1.6.2               fitdistrplus_1.1-11    purrr_1.0.1           
 [82] RANN_2.6.1             nlme_3.1-162           pbapply_1.7-0         
 [85] future_1.32.0          whisker_0.4.1          mime_0.12             
 [88] compiler_4.2.2         rstudioapi_0.14        plotly_4.10.1         
 [91] png_0.1-8              ggsignif_0.6.4         spatstat.utils_3.0-2  
 [94] tibble_3.2.1           bslib_0.4.2            stringi_1.7.12        
 [97] highr_0.10             lattice_0.21-8         Matrix_1.5-4          
[100] vctrs_0.6.2            pillar_1.9.0           lifecycle_1.0.3       
[103] spatstat.geom_3.1-0    lmtest_0.9-40          jquerylib_0.1.4       
[106] RcppAnnoy_0.0.20       data.table_1.14.8      cowplot_1.1.1         
[109] bitops_1.0-7           irlba_2.3.5.1          httpuv_1.6.9          
[112] patchwork_1.1.2        R6_2.5.1               promises_1.2.0.1      
[115] KernSmooth_2.23-20     gridExtra_2.3          parallelly_1.35.0     
[118] codetools_0.2-19       MASS_7.3-59            rprojroot_2.0.3       
[121] withr_2.5.0            sctransform_0.3.5      GenomeInfoDbData_1.2.9
[124] parallel_4.2.2         grid_4.2.2             tidyr_1.3.0           
[127] rmarkdown_2.21         carData_3.0-5          Rtsne_0.16            
[130] git2r_0.32.0           spatstat.explore_3.1-0 shiny_1.7.4           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
