<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DE analysis on scRNA data with poisson GLMM</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEanalysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/C-HW/DEanalysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DE analysis on scRNA data with poisson
GLMM</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DEanalysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomchwDEanalysistreeeeb5d79df9d42c0f2da67d83243a6880f3fb8170targetblankeeb5d79a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/c-hw/DEanalysis/tree/eeb5d79df9d42c0f2da67d83243a6880f3fb8170" target="_blank">eeb5d79</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomchwDEanalysistreeeeb5d79df9d42c0f2da67d83243a6880f3fb8170targetblankeeb5d79a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/c-hw/DEanalysis/tree/eeb5d79df9d42c0f2da67d83243a6880f3fb8170" target="_blank">eeb5d79</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  .Rhistory
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/.Rapp.history
    Untracked:  analysis/.Rhistory
    Untracked:  analysis/Bcells.Rmd
    Untracked:  analysis/CD14+ Monocytes.Rmd
    Untracked:  analysis/analysis on Kang.Rmd
    Untracked:  analysis/data2.Rmd
    Untracked:  analysis/data_clusters.Rmd
    Untracked:  analysis/first-analysis.Rmd
    Untracked:  analysis/group12_13.Rmd
    Untracked:  analysis/group12_19.Rmd
    Untracked:  analysis/group2_19.Rmd
    Untracked:  analysis/group8_17-2_19.Rmd
    Untracked:  analysis/group8_17.Rmd
    Untracked:  analysis/methods_details.Rmd
    Untracked:  analysis/new_criteria.Rmd
    Untracked:  data/.Rhistory
    Untracked:  data/10X_Kang_DEresult.RData
    Untracked:  data/10X_inputdata.RData
    Untracked:  data/10X_inputdata_DEresult.RData
    Untracked:  data/10X_inputdata_cpm.RData
    Untracked:  data/10X_inputdata_integrated.RData
    Untracked:  data/10X_inputdata_lognorm.RData
    Untracked:  data/10Xdata_annotate.rds
    Untracked:  data/Bcells.Rmd
    Untracked:  data/Bcellsce.rds
    Untracked:  data/data2sce.RData
    Untracked:  data/permutation.RData
    Untracked:  data/vstcounts.Rdata

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/9daafa50ed16dbc150a5f207a0151d29ae5c25cc/docs/index.html" target="_blank">9daafa5</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-30
</td>
<td>
index small adjustment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/59b08c200b9c52499eb93ac1d3db385ede971a1f/docs/index.html" target="_blank">59b08c2</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-29
</td>
<td>
update index, FD permuation, plots axes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/e5cde057fc370caf99ec755efbdcf2e91d962225/docs/index.html" target="_blank">e5cde05</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-13
</td>
<td>
update index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/fc4ab47aef45358f9742d154ee4e6b3963876617/docs/index.html" target="_blank">fc4ab47</a>
</td>
<td>
C-HW
</td>
<td>
2023-08-03
</td>
<td>
data2 results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/5ee946038b6565ced58775dd29d83d7e79024e83/docs/index.html" target="_blank">5ee9460</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-27
</td>
<td>
update index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/949c7601790e106ec05a03550fcd0a9c9d5f9d1e/docs/index.html" target="_blank">949c760</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-27
</td>
<td>
data2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/b688ebf5fd852d4a570b619eb3adb36d0b50561f/docs/index.html" target="_blank">b688ebf</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-13
</td>
<td>
updatd index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/4da4a48eb74a20df5416741f3df5fec69133e6e6/docs/index.html" target="_blank">4da4a48</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-13
</td>
<td>
update index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/35f04ca27d51636c1fdf7db1b380a3109ad2568b/docs/index.html" target="_blank">35f04ca</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-12
</td>
<td>
add 12_19
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/366cd53204c6da534d7b31df1350afc7cc2f8507/analysis/index.Rmd" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/366cd53204c6da534d7b31df1350afc7cc2f8507/docs/index.html" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/3b1c8ca23e5cf81840f1c20f0171842c96960cea/docs/index.html" target="_blank">3b1c8ca</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update index page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/95be1229d96483e02a29434ab22951bcf2db2c2b/analysis/index.Rmd" target="_blank">95be122</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update 2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/95be1229d96483e02a29434ab22951bcf2db2c2b/docs/index.html" target="_blank">95be122</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update 2_19
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/2107af314e94546d55eb2980a145072271b951af/analysis/index.Rmd" target="_blank">2107af3</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-24
</td>
<td>
add methods_details
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/9a7a5694008de6ed6ea81ba0bc83db6ea833df44/analysis/index.Rmd" target="_blank">9a7a569</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
correct link
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/9a7a5694008de6ed6ea81ba0bc83db6ea833df44/docs/index.html" target="_blank">9a7a569</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
correct link
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/641a5d8fc7ffeee15db491690cb3cc1d10a624a9/analysis/index.Rmd" target="_blank">641a5d8</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add link to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/641a5d8fc7ffeee15db491690cb3cc1d10a624a9/docs/index.html" target="_blank">641a5d8</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add link to index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/13d726dde0214c2d74705c21397e8bb2f49f8b6f/analysis/index.Rmd" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add DE results on different groups
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/13d726dde0214c2d74705c21397e8bb2f49f8b6f/docs/index.html" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add DE results on different groups
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/fc9f4b6a78f05402b94483374fba17363245b131/analysis/index.Rmd" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add new_criteria
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/fc9f4b6a78f05402b94483374fba17363245b131/docs/index.html" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add new_criteria
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/75869536c17b66292074ac0e2e25deaf985d149e/docs/index.html" target="_blank">7586953</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
<td>
add data_clusters
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/703eb541802b4378446d9699f4ae5ce7a1b809f3/analysis/index.Rmd" target="_blank">703eb54</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-09
</td>
<td>
create workflowr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/703eb541802b4378446d9699f4ae5ce7a1b809f3/docs/index.html" target="_blank">703eb54</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-09
</td>
<td>
create workflowr
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>Differential expression (DE) analysis in single-cell transcriptomics
reveals cell-type-specific responses. Recent studies have raised
concerns about current methods used to identify differentially expressed
genes in this context.</p>
<ul>
<li><p>scRNA sequencing provides the absolute abundances of RNA
molecules in single cells, but normalization - a pre-processing step
inherited from the bulk RNA-seq era - reduces this information and
returns data as relative abundances, which may mask important
differences among cell types</p></li>
<li><p>The majority of single-cell DE analysis methods are susceptible
to generating false discoveries. This is mainly due to the lack of
accounting for variations between biological replicates, commonly
referred to as “donor effects”.</p>
<ul>
<li>Batch effects are often estimated from leading principal components,
representing a consensus from most genes.</li>
<li>Pseudo-bulk analysis ignores within-sample heterogeneity by treating
donor effects as a fixed effect and assuming that each cell from the
same donor is equally affected.</li>
</ul></li>
<li><p>Clustering and DE analysis are different problems. The current
commonly workflow works well in clustering, but cannot guarantee success
in downstream analysis.</p></li>
<li><p>Excessive zeros are usually considered as “drop-outs”, while they
are actually informative in cell-type heterogeneity. Ignoring zeros in
single-cell gene expression data discards valuable information for any
analysis.</p></li>
</ul>
<p>We provide a generalized linear mixed model framework (GLMM) to
detect differentially expressed genes (DEGs) between two given cell
types. The model takes donor-specific variations as random effects, and
uses raw UMI counts to prevent biases in DE analysis.</p>
</div>
<div id="glmm-for-de-analysis" class="section level2">
<h2>GLMM for DE analysis</h2>
<div id="poisson-glmm" class="section level3">
<h3>Poisson GLMM</h3>
<p>For each count <span class="math inline">\(X_{cgk}\)</span> sampled
from cell <span class="math inline">\(c\)</span>, donor <span
class="math inline">\(k\)</span>, and gene <span
class="math inline">\(g\)</span>,</p>
<p><span class="math display">\[
\begin{aligned}
    X_{cgk}|\lambda_{cgk} &amp; \sim Poisson(\lambda_{cgk})\\
    \log \lambda_{cgk} &amp; = \mu_g + X_c{\beta_g} + \epsilon_{gk}\\
\end{aligned}
\]</span> where <span class="math inline">\(X_c\)</span> is the
indicator for different cell types, and <span
class="math inline">\(\epsilon_{gk}\sim N(0,\sigma_{g}^2)\)</span>
represents the random effects for donor <span
class="math inline">\(k\)</span>. Our goal is to test <span
class="math inline">\(H_0: \beta_g = 0\)</span>. Here <span
class="math inline">\(e^{\beta_g}\)</span> represents the fold change of
gene <span class="math inline">\(g\)</span> between two cell types.</p>
</div>
<div id="binomial-glmm" class="section level3">
<h3>Binomial GLMM</h3>
<p><span class="math display">\[
\begin{aligned}
    \mathbb{1}_{X_{cgk}=0}|p_{cgk} &amp; \sim Bernoulli(p_{cgk})\\
    \log \frac{p_{cgk}}{1-p_{cgk}} &amp; = \mu_g + X_c\beta_{g} +
\epsilon_{gk}\\
\end{aligned}
\]</span> where <span class="math inline">\(X_c\)</span> is the
indicator for different cell types, and <span
class="math inline">\(\epsilon_{gk}\sim N(0,\sigma_{g}^2)\)</span>
represents the random effects for donor <span
class="math inline">\(k\)</span>. Our goal is to test <span
class="math inline">\(H_0: \beta_g = 0\)</span>.</p>
<p><a href="new_criteria.html">new criteria</a></p>
</div>
</div>
</div>
<div id="application" class="section level1">
<h1>Application</h1>
<div id="data-summary" class="section level2">
<h2>Data summary</h2>
<p><a href="data_clusters.html">Data</a></p>
</div>
<div id="analysis-and-methods-comparison" class="section level2">
<h2>Analysis and Methods comparison</h2>
<p><a href="methods_details.html">Methods details</a><br />
<a href="group12_13.html">group12_13</a><br />
<a href="group2_19.html">group2_19</a><br />
<a href="group8_17-2_19.html">group8_17&amp;2_19</a></p>
</div>
</div>
<div id="simulation" class="section level1">
<h1>Simulation</h1>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
