<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>The curses of performing differential expression analysis using single-cell data</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEanalysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/C-HW/DEanalysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">The curses of performing differential
expression analysis using single-cell data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DEanalysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCHWDEanalysistree6480f5cf64234aeddae4d4afdc5efa966447e033targetblank6480f5ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/C-HW/DEanalysis/tree/6480f5cf64234aeddae4d4afdc5efa966447e033" target="_blank">6480f5c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCHWDEanalysistree6480f5cf64234aeddae4d4afdc5efa966447e033targetblank6480f5ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/C-HW/DEanalysis/tree/6480f5cf64234aeddae4d4afdc5efa966447e033" target="_blank">6480f5c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.Rhistory

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  analysis/analysis_humanspine.Rmd
    Untracked:  analysis/simulation_donor_effect.Rmd
    Untracked:  data/.DS_Store
    Untracked:  data/10X_DEresult_update.RData
    Untracked:  data/10X_DEresult_update2.RData
    Untracked:  data/10X_Kang_DEresult.RData
    Untracked:  data/10X_inputdata.RData
    Untracked:  data/10X_inputdata_DEresult.RData
    Untracked:  data/10X_inputdata_cpm.RData
    Untracked:  data/10X_inputdata_integrated.RData
    Untracked:  data/10X_inputdata_lognorm.RData
    Untracked:  data/10Xdata_annotate.rds
    Untracked:  data/Bcells.Rmd
    Untracked:  data/Bcellsce.rds
    Untracked:  data/Kang_DEresult.RData
    Untracked:  data/Kang_data.RData
    Untracked:  data/fallopian_tubes.RData
    Untracked:  data/fallopian_tubes_all.RData
    Untracked:  data/human/
    Untracked:  data/human_spine.RData
    Untracked:  data/human_spine_DEresult.RData
    Untracked:  data/mouse/
    Untracked:  data/permutation.RData
    Untracked:  data/permutation13.RData
    Untracked:  data/permutation2.RData
    Untracked:  data/splatter_simulation.RData
    Untracked:  data/splatter_simulation1.RData
    Untracked:  data/splatter_simulation2.RData
    Untracked:  data/splatter_simulation3.RData
    Untracked:  data/vstcounts.Rdata
    Untracked:  figure/

Unstaged changes:
    Modified:   analysis/.Rhistory
    Modified:   analysis/CD14+-Monocytes.Rmd
    Modified:   code/DE_methods.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/6480f5cf64234aeddae4d4afdc5efa966447e033/analysis/index.Rmd" target="_blank">6480f5c</a>
</td>
<td>
C-HW
</td>
<td>
2024-10-19
</td>
<td>
add human spine
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/eaef8119d9e03d800655722c807318e31fbdc9bb/docs/index.html" target="_blank">eaef811</a>
</td>
<td>
C-HW
</td>
<td>
2024-10-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/bfc62003c66743cc818e33fa8f52f8ddca11ab05/analysis/index.Rmd" target="_blank">bfc6200</a>
</td>
<td>
C-HW
</td>
<td>
2024-10-14
</td>
<td>
workflowr::wflow_publish("index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/42769e7b58dcf673c13d112f2c24d24d23a775a2/docs/index.html" target="_blank">42769e7</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/da950e4f396a01418480cf5b17aa4c6bd3979ee5/analysis/index.Rmd" target="_blank">da950e4</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
remove about data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/76c848acd970be1a41c883f8f293ef6a9d880a92/docs/index.html" target="_blank">76c848a</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/cf1ad3a91c75c5f51e6646b133645c633ede8619/analysis/index.Rmd" target="_blank">cf1ad3a</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
revise order on index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/e10a2568a2321eaaca544bd14ddeb2f1e7f63f91/docs/index.html" target="_blank">e10a256</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/2b477ada3562b30bbf69d8d65dfad5511098cf5e/analysis/index.Rmd" target="_blank">2b477ad</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
revise order on index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/2fba81eacdd41adc3e2b1e97eb8b4cf3974fb774/docs/index.html" target="_blank">2fba81e</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/ea89e68e85f58f8d8303c79a98ec8842fd6ed026/analysis/index.Rmd" target="_blank">ea89e68</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-15
</td>
<td>
revise order on index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/463f4a9147379ec4adb5b6b0feb35ef2c1dc3302/docs/index.html" target="_blank">463f4a9</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/59bf3274985fb0d2d4cba2fa9acca891258e1942/docs/index.html" target="_blank">59bf327</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/d05c0685b3d81cf870d87b1354f2ffa8079747e1/docs/index.html" target="_blank">d05c068</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/802697b2457a506f85ae691f91df323cbdca630a/docs/index.html" target="_blank">802697b</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
index html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/38036979db676956659d028827462d615c9f4cf1/analysis/index.Rmd" target="_blank">3803697</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
upload rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/38036979db676956659d028827462d615c9f4cf1/docs/index.html" target="_blank">3803697</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
upload rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/a0d06fec214c56660c774327d33d6f996dc83216/analysis/index.Rmd" target="_blank">a0d06fe</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
recover index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/a0d06fec214c56660c774327d33d6f996dc83216/docs/index.html" target="_blank">a0d06fe</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
recover index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/828d72e4651a30b4bf79e14248ab40bc831a29b7/docs/index.html" target="_blank">828d72e</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
update index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/9daafa50ed16dbc150a5f207a0151d29ae5c25cc/docs/index.html" target="_blank">9daafa5</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-30
</td>
<td>
index small adjustment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/59b08c200b9c52499eb93ac1d3db385ede971a1f/docs/index.html" target="_blank">59b08c2</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-29
</td>
<td>
update index, FD permuation, plots axes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/e5cde057fc370caf99ec755efbdcf2e91d962225/docs/index.html" target="_blank">e5cde05</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-13
</td>
<td>
update index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/fc4ab47aef45358f9742d154ee4e6b3963876617/docs/index.html" target="_blank">fc4ab47</a>
</td>
<td>
C-HW
</td>
<td>
2023-08-03
</td>
<td>
data2 results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/5ee946038b6565ced58775dd29d83d7e79024e83/docs/index.html" target="_blank">5ee9460</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-27
</td>
<td>
update index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/949c7601790e106ec05a03550fcd0a9c9d5f9d1e/docs/index.html" target="_blank">949c760</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-27
</td>
<td>
data2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/b688ebf5fd852d4a570b619eb3adb36d0b50561f/docs/index.html" target="_blank">b688ebf</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-13
</td>
<td>
updatd index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/4da4a48eb74a20df5416741f3df5fec69133e6e6/docs/index.html" target="_blank">4da4a48</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-13
</td>
<td>
update index.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/35f04ca27d51636c1fdf7db1b380a3109ad2568b/docs/index.html" target="_blank">35f04ca</a>
</td>
<td>
C-HW
</td>
<td>
2023-07-12
</td>
<td>
add 12_19
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/366cd53204c6da534d7b31df1350afc7cc2f8507/analysis/index.Rmd" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/366cd53204c6da534d7b31df1350afc7cc2f8507/docs/index.html" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/3b1c8ca23e5cf81840f1c20f0171842c96960cea/docs/index.html" target="_blank">3b1c8ca</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update index page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/95be1229d96483e02a29434ab22951bcf2db2c2b/analysis/index.Rmd" target="_blank">95be122</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update 2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/95be1229d96483e02a29434ab22951bcf2db2c2b/docs/index.html" target="_blank">95be122</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-25
</td>
<td>
update 2_19
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/2107af314e94546d55eb2980a145072271b951af/analysis/index.Rmd" target="_blank">2107af3</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-24
</td>
<td>
add methods_details
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/9a7a5694008de6ed6ea81ba0bc83db6ea833df44/analysis/index.Rmd" target="_blank">9a7a569</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
correct link
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/9a7a5694008de6ed6ea81ba0bc83db6ea833df44/docs/index.html" target="_blank">9a7a569</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
correct link
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/641a5d8fc7ffeee15db491690cb3cc1d10a624a9/analysis/index.Rmd" target="_blank">641a5d8</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add link to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/641a5d8fc7ffeee15db491690cb3cc1d10a624a9/docs/index.html" target="_blank">641a5d8</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add link to index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/13d726dde0214c2d74705c21397e8bb2f49f8b6f/analysis/index.Rmd" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add DE results on different groups
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/13d726dde0214c2d74705c21397e8bb2f49f8b6f/docs/index.html" target="_blank">13d726d</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add DE results on different groups
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/fc9f4b6a78f05402b94483374fba17363245b131/analysis/index.Rmd" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add new_criteria
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/fc9f4b6a78f05402b94483374fba17363245b131/docs/index.html" target="_blank">fc9f4b6</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-18
</td>
<td>
add new_criteria
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/75869536c17b66292074ac0e2e25deaf985d149e/docs/index.html" target="_blank">7586953</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-11
</td>
<td>
add data_clusters
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/C-HW/DEanalysis/blob/703eb541802b4378446d9699f4ae5ce7a1b809f3/analysis/index.Rmd" target="_blank">703eb54</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-09
</td>
<td>
create workflowr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/C-HW/DEanalysis/703eb541802b4378446d9699f4ae5ce7a1b809f3/docs/index.html" target="_blank">703eb54</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-09
</td>
<td>
create workflowr
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="motivation" class="section level1">
<h1>Motivation</h1>
<p>Differential expression (DE) analysis in single-cell transcriptomics
reveals cell-type-specific responses. Recent studies have raised
concerns about current methods used to identify differentially expressed
genes in this context.</p>
<ul>
<li><p>scRNA sequencing provides the absolute abundances of RNA
molecules in single cells, but normalization - a pre-processing step
inherited from the bulk RNA-seq era - reduces this information and
returns data as relative abundances, which may mask important
differences among cell types</p></li>
<li><p>The majority of single-cell DE analysis methods are susceptible
to generating false discoveries. This is mainly due to the lack of
accounting for variations between biological replicates, commonly
referred to as “donor effects”.</p>
<ul>
<li>Batch effects are often estimated from leading principal components,
representing a consensus from most genes.</li>
<li>Pseudo-bulk analysis ignores within-sample heterogeneity by treating
donor effects as a fixed effect and assuming that each cell from the
same donor is equally affected.</li>
</ul></li>
<li><p>Clustering and DE analysis are different problems. The current
commonly workflow works well in clustering, but cannot guarantee success
in downstream analysis.</p></li>
<li><p>Excessive zeros are usually considered as “drop-outs”, while they
are actually informative in cell-type heterogeneity. Ignoring zeros in
single-cell gene expression data discards valuable information for any
analysis.</p></li>
</ul>
<p>We provided a generalized linear mixed model framework (<a
href="https://github.com/C-HW/LEMUR">LEMUR</a>: Linear mixed model-based
differential Expression analysis accounting for Multiple samples Using
Random effects) to detect differentially expressed genes (DEGs) between
two given cell types. The model takes donor-specific variations as
random effects, and uses raw UMI counts to prevent biases in DE
analysis.</p>
<p><img src="./docs/figure/panel3.png" /></p>
</div>
<div id="lemur-framework" class="section level1">
<h1>LEMUR framework</h1>
<p>By default, we excluded any genes detected in fewer than 5% cells in
the compared groups from differential testing. The GLMMs were
implemented with glmmPQL function of the MASS package. We calculated
adjusted p-values by using Benjamini-Hochberg correction. Each model
fitting was applied on one gene and the two compared groups.</p>
<div id="poisson-glmm" class="section level2">
<h2>Poisson-glmm</h2>
<p>For each count <span class="math inline">\(X_{cgk}\)</span> sampled
from cell <span class="math inline">\(c\)</span>, donor <span
class="math inline">\(k\)</span>, and gene <span
class="math inline">\(g\)</span>,</p>
<p><span class="math display">\[
\begin{aligned}
    X_{cgk}|\lambda_{cgk} &amp; \sim Poisson(\lambda_{cgk})\\
    \log \lambda_{cgk} &amp; = \mu_g + X_c{\beta_g} + \epsilon_{gk}\\
\end{aligned}
\]</span> where <span class="math inline">\(X_c\)</span> is the
indicator for group-of-interest (cell types, conditions, regions), and
<span class="math inline">\(\epsilon_{gk}\sim N(0,\sigma_{g}^2)\)</span>
represents the random effects for donor <span
class="math inline">\(k\)</span>. Our goal is to test <span
class="math inline">\(H_0: \beta_g = 0\)</span>. Here <span
class="math inline">\(e^{\beta_g}\)</span> represents the fold change of
gene <span class="math inline">\(g\)</span> between
group-of-interest.</p>
</div>
<div id="binomial-glmm" class="section level2">
<h2>Binomial-glmm</h2>
<p><span class="math display">\[
\begin{aligned}
    \mathbb{1}_{X_{cgk}=0}|p_{cgk} &amp; \sim Bernoulli(p_{cgk})\\
    \log \frac{p_{cgk}}{1-p_{cgk}} &amp; = \mu_g + X_c\beta_{g} +
\epsilon_{gk}\\
\end{aligned}
\]</span> where <span class="math inline">\(X_c\)</span> is the
indicator for group-of-interest (cell types, conditions, regions), and
<span class="math inline">\(\epsilon_{gk}\sim N(0,\sigma_{g}^2)\)</span>
represents the random effects for donor <span
class="math inline">\(k\)</span>. Our goal is to test <span
class="math inline">\(H_0: \beta_g = 0\)</span>.</p>
<p>For both methods, we provided “log2 fold change” computed by <span
class="math inline">\(\log_2(\beta_g)\)</span> . In Poisson-glmm, this
estimate indicates the increment of <span
class="math inline">\(\log_2(\lambda_2)\)</span> against <span
class="math inline">\(\log_2(\lambda_1)\)</span>, which is the
conventional log2 fold change. However, this term in Binomial-glmm
doesn’t represent the same meaning. It is the difference between
logit(<span class="math inline">\(p_2\)</span>) and logit(<span
class="math inline">\(p_1\)</span>). The p-value and BH adjusted p-value
are provided.</p>
</div>
</div>
<div id="benchmarked-methods" class="section level1">
<h1>Benchmarked methods</h1>
<p>In this project, we compared a few methods performing the DE analysis
results. Our comparison encompassed Poisson-glmm and Binomial-glmm from
the new paradigm, as well as pseudo-bulk approaches including DESeq2 and
edgeR. Additionally, we assessed the performance of single cell specific
tools including MAST, Wilcox in Seurat, and linear mixed models in
Muscat. More details can be found <a
href="methods_details.html">here</a>.</p>
</div>
<div id="new-criteria-on-determining-degs" class="section level1">
<h1>New criteria on determining DEGs</h1>
<p>We proposed new criteria that based on the convention and also the
gene mean and the difference in mean. If the log2 gene mean in two
groups is lower than a certain value (-2.25 in case study 1) and the
log2 mean difference is smaller than a threshold (-1 in case study 1),
the genes would not be considered as a DEGs. These can also be used as a
filter before any DE analysis to speed up the computation. Both of these
criteria are adjustable, depending on the dataset’s performance and
characteristics. An examination in heatmaps and mean difference against
mean plot in advanced can be helpful to determine the thresholds when
analyzing a new dataset. More details can be found <a
href="new_criteria.html">here</a>.</p>
</div>
<div id="application" class="section level1">
<h1>Application</h1>
<div id="case-study-1-fallopian-tubes-immune-cells"
class="section level2">
<h2>Case study 1 (Fallopian tubes immune cells)</h2>
<p>In case study 1, a 10X scRNA-seq dataset of post-menopausal fallopian
tubes, with 57,182 cells sourced from five donors, covering 29,382 genes
is analyzed. The 20 clusters are obtained via HIPPO algorithm. There is
no pre-filtering procedure applied on this dataset, except for built-in
filtering steps in each method. We use sctransform to get the VST data,
and the integration workflow (v4) provided by Seurat to obtain the
integrated data.</p>
<ul>
<li><p><a href="fallopian_tubes_overview.html">Data summary</a></p></li>
<li><p>Analyses</p>
<ul>
<li><a href="group12_13.html">Group12_13</a><br />
</li>
<li><a href="group2_19.html">Group2_19</a><br />
</li>
<li><a href="group8_17-2_19.html">Group8_17&amp;2_19</a></li>
<li><a href="group12_19.html">Group12_19</a></li>
<li><a href="group8_17.html">Group8_17</a></li>
</ul></li>
</ul>
</div>
<div id="case-study-2-human-spine-cells" class="section level2">
<h2>Case study 2 (Human spine cells)</h2>
<p>In case study 2, the dataset contains scRNA-seq data from 9 patients
and sequenced in 16 slides. The dataset consists of a total of 48644
cells. After removing ambiguous genes and only keeping common genes
across all samples, it includes sequencing data for 8092 genes.</p>
<ul>
<li><p><a href="human_spine_overview.html">Data summary</a></p></li>
<li><p>Analyses</p>
<ul>
<li><a href="Dors_Horn_Vent_Lat_White.html">Dorsal Horn vs. entral
Lateral White</a></li>
</ul></li>
</ul>
</div>
<div id="case-study-3-lupus-patients-with-treatment"
class="section level2">
<h2>Case study 3 (Lupus patients with treatment)</h2>
<p>In case study 3, the dataset contains 10X droplet-based scRNA-seq
data from 8 Lupus patients obtained before and after 6h-treatment with
IFN-β. After removing undetected genes and lowly expressed genes (less
than 10 cells expressing more than 1), the dataset consists of 29065
cells and 7661 genes. The integrated data is replaced by
log2-transformed normalized expression values obtained via
computeLibrarayFactors and logNormCounts functions in Muscat.</p>
<ul>
<li><p><a href="Kang_overview.html">Data summary</a></p></li>
<li><p>Analyses</p>
<ul>
<li><a href="Bcells.html">B cells</a></li>
<li><a href="CD14+-Monocytes.html">CD14+ Monocytes</a></li>
</ul></li>
</ul>
</div>
<div id="permutation-analysis" class="section level2">
<h2>Permutation analysis</h2>
<p>To examine the p-value calibration in real data, we did a permutation
analysis on group-of-interest within a null dataset. The results can be
found <a href="FD_analysis.html">here</a>.</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
