<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chih-Hsuan Wu" />


<title>Methods details</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEanalysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/C-HW/DEanalysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Methods details</h1>
<h4 class="author">Chih-Hsuan Wu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DEanalysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomchwDEanalysistree549857a183a661061d19fa791eec5b0b1c846d07targetblank549857aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/c-hw/DEanalysis/tree/549857a183a661061d19fa791eec5b0b1c846d07" target="_blank">549857a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomchwDEanalysistree549857a183a661061d19fa791eec5b0b1c846d07targetblank549857aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/c-hw/DEanalysis/tree/549857a183a661061d19fa791eec5b0b1c846d07" target="_blank">549857a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  .Rhistory
    Untracked:  data/.Rhistory
    Untracked:  data/10X_Kang_DEresult.RData
    Untracked:  data/10X_inputdata.RData
    Untracked:  data/10X_inputdata_DEresult.RData
    Untracked:  data/10X_inputdata_cpm.RData
    Untracked:  data/10X_inputdata_integrated.RData
    Untracked:  data/10X_inputdata_lognorm.RData
    Untracked:  data/10Xdata_annotate.rds
    Untracked:  data/Bcells.Rmd
    Untracked:  data/Bcellsce.rds
    Untracked:  data/data2sce.RData
    Untracked:  data/permutation.RData
    Untracked:  data/vstcounts.Rdata

Unstaged changes:
    Modified:   code/DE_methods.R
    Modified:   code/functions_in_rmd.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/methods_details.Rmd</code>) and
HTML (<code>docs/methods_details.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/c-hw/DEanalysis/blob/38036979db676956659d028827462d615c9f4cf1/analysis/methods_details.Rmd" target="_blank">3803697</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
upload rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/38036979db676956659d028827462d615c9f4cf1/docs/methods_details.html" target="_blank">3803697</a>
</td>
<td>
C-HW
</td>
<td>
2023-12-01
</td>
<td>
upload rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/59872e4a32e69f281ee862caae8d5cd188dd311b/docs/methods_details.html" target="_blank">59872e4</a>
</td>
<td>
C-HW
</td>
<td>
2023-11-13
</td>
<td>
update methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/366cd53204c6da534d7b31df1350afc7cc2f8507/docs/methods_details.html" target="_blank">366cd53</a>
</td>
<td>
C-HW
</td>
<td>
2023-06-06
</td>
<td>
add group8_17&amp;2_19
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/c-hw/DEanalysis/2107af314e94546d55eb2980a145072271b951af/docs/methods_details.html" target="_blank">2107af3</a>
</td>
<td>
C-HW
</td>
<td>
2023-05-24
</td>
<td>
add methods_details
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In this project, we compare a few methods performing the DE analysis
results. Each of them might use different input data ,different
statistical model and different FDR control on p-values.</p>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="5%" />
<col width="6%" />
<col width="28%" />
<col width="15%" />
<col width="7%" />
<col width="9%" />
<col width="7%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"></th>
<th align="center">Poisson-glmm</th>
<th align="center">Binomial-glmm</th>
<th align="center">Pb-DESeq2</th>
<th align="center">Pb-edgeR</th>
<th align="center">MAST</th>
<th align="center">Wilcox</th>
<th align="center">MMvst</th>
<th align="center">MMpoisson</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Package</td>
<td align="center">Our method</td>
<td align="center">Our method</td>
<td align="center">Muscat</td>
<td align="center">Muscat</td>
<td align="center">MAST</td>
<td align="center">Seurat</td>
<td align="center">Muscat</td>
<td align="center">Muscat</td>
</tr>
<tr class="even">
<td align="center">Input</td>
<td align="center">UMI</td>
<td align="center">zero proportion</td>
<td align="center">UMI</td>
<td align="center">CPM</td>
<td align="center">CPM</td>
<td align="center">Integrated/Log normalized</td>
<td align="center">VST</td>
<td align="center">UMI</td>
</tr>
<tr class="odd">
<td align="center">Model base</td>
<td align="center">Poisson glmm</td>
<td align="center">Binomial glmm</td>
<td align="center">Negative binomial model</td>
<td align="center">Negative binomial model</td>
<td align="center">Zero-inflated model</td>
<td align="center">Rank-sum test</td>
<td align="center">Linear mixed model</td>
<td align="center">Poisson glmm</td>
</tr>
<tr class="even">
<td align="center">Normalization</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">V</td>
<td align="center">V</td>
<td align="center">V</td>
<td align="center">V</td>
<td align="center">V</td>
<td align="center">V</td>
</tr>
<tr class="odd">
<td align="center">Normalization method</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">Median of ratio size factor, variance stabilizing
transformation in model</td>
<td align="center">In data, trimmed mean of M values (TMM) in model</td>
<td align="center">In data</td>
<td align="center">In data</td>
<td align="center">In data</td>
<td align="center">Library size factor as offset</td>
</tr>
</tbody>
</table>
<div id="poisson-glmm" class="section level1">
<h1>Poisson GLMM</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing raw counts generated
by 10X protocol<br />
</li>
<li>cellgroups: A vector labeling the cell groups<br />
</li>
<li>repgroups: A vector labeling the replicates (donors)<br />
</li>
<li>freq_expressed: A threshold value of gene expression frequency
(default = <span class="math inline">\(0.05\)</span>)</li>
</ul></li>
<li>Details:
<ul>
<li><p>Only genes that pass the threshold for gene expression frequency
will be considered as inputs.</p></li>
<li><p>For each gene, we run a poisson glmm method (glmmPQL) on the raw
counts with cellgroups as fixed effect and repgroups as random effect
<span class="math display">\[
\begin{aligned}
X_{cgk}|\lambda_{cgk} &amp; \sim Poisson(\lambda_{cgk})\\
\log \lambda_{cgk} &amp; = \mu_g + X_c{\beta_g} + \epsilon_{gk}\\
\end{aligned}
\]</span></p></li>
<li><p>If the algorithm doesn’t converge, the gene will be
excluded.</p></li>
</ul></li>
<li>Output:
<ul>
<li>mu: The base line log mean count for the first cell group</li>
<li>beta_cellgroup: The coefficient of cellgroups</li>
<li>log2FC: log2 fold change <span
class="math inline">\(\log_2(e^{\beta_g})\)</span> of counts between the
two groups. Positive values indicate that the gene is more expressed in
the second group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure</li>
<li>REvariation:</li>
<li>FEvariation:</li>
<li>RESvariation:</li>
<li>hits: Indicating whether the gene is a DEG based on the <a
href="new_criteria.html">new criteria</a></li>
</ul></li>
</ul>
</div>
<div id="binomial-glmm" class="section level1">
<h1>Binomial GLMM</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing raw counts generated
by 10X protocol<br />
</li>
<li>cellgroups: A vector labeling the cell groups<br />
</li>
<li>repgroups: A vector labeling the replicates (donors)<br />
</li>
<li>freq_expressed: A threshold value of gene expression frequency
(default = <span class="math inline">\(0.05\)</span>)</li>
</ul></li>
<li>Details:
<ul>
<li><p>Only genes that pass the threshold for gene expression frequency
will be considered as inputs.</p></li>
<li><p>We take the zero proportion of each gene as the response in the
binomial model. (<span class="math inline">\(1\)</span> if the read
count is zero; otherwise <span
class="math inline">\(0\)</span>.)</p></li>
<li><p>For each gene, we run a binomial glmm method (glmmPQL) on the
zero proportion with cellgroups as fixed effect and repgroups as random
effect <span class="math display">\[
\begin{aligned}
\mathbb{1}_{X_{cgk}=0}|p_{cgk} &amp; \sim Bernoulli(p_{cgk})\\
\log \frac{p_{cgk}}{1-p_{cgk}} &amp; = \mu_g + X_c\beta_{g} +
\epsilon_{gk}\\
\end{aligned}
\]</span></p></li>
<li><p>If the algorithm doesn’t converge, the gene will be
excluded.</p></li>
</ul></li>
<li>Output:
<ul>
<li>mu: The base line of logit zero proportion for the first cell
group</li>
<li>beta_cellgroup: The coefficient of cellgroups</li>
<li>log2FC: In Binomial GLMM, <span
class="math inline">\(\log_2(e^{\beta_g})\)</span> represents the log2
odds ratio change between group1 and group2. Positive values indicate
that the gene is more expressed in the second group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure</li>
<li>hits: Indicating whether the gene is a DEG based on the <a
href="new_criteria.html">new criteria</a></li>
</ul></li>
</ul>
</div>
<div id="pseudobulk-deseq2" class="section level1">
<h1>Pseudobulk DEseq2</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing raw counts generated
by 10X protocol<br />
</li>
<li>group_id: A feature labeling the cell groups or states</li>
<li>sample_id: A feature labeling the replicates (donors)</li>
</ul></li>
<li>Details:
<ul>
<li>Aggregate the counts within same donor and same cell group.</li>
<li>Run Muscat::pbDS(pb, method = “DESeq2”) on the pseudobulk
counts</li>
<li>DESeq2 performs an internal normalization and use the median of
ratios as the size factor.</li>
<li>DESeq2 fits negative binomial generalized linear models for each
gene and uses the Wald test for significance testing.</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/vignettes/muscat/inst/doc/analysis.html">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>log2FC: log2 fold change of counts between the two groups. Positive
values indicate that the gene is more expressed in the first group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if BH is smaller
than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>
<div id="pseudobulk-edger" class="section level1">
<h1>Pseudobulk edgeR</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing CPM counts</li>
<li>cellgroups: A vector labeling the cell groups<br />
</li>
<li>repgroups: A vector labeling the replicates (donors) (NA if without
donor effect)</li>
</ul></li>
<li>Details:
<ul>
<li>Aggregate the CPM counts within same donor and same cell group.</li>
<li>Run Muscat::pbDS(pb, method = “edgeR”) on the pseudobulk counts</li>
<li>EdgeR uses TMM method to compute normalization factors which are
multiplied by the library size to yield the effective library size.</li>
<li>EdgeR fits negative binomial generalized linear models.</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/vignettes/muscat/inst/doc/analysis.html">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>log2FC: log2 fold change of counts between the two groups. Positive
values indicate that the gene is more expressed in the first group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if BH is smaller
than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>
<div id="mast" class="section level1">
<h1>MAST</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing raw counts generated
by 10X protocol</li>
<li>cellgroups: A vector labeling the cell groups<br />
</li>
<li>repgroups: A vector labeling the replicates (donors) (NA if without
donor effect)</li>
<li>freq_expressed: A threshold value of gene expression frequency
(default = <span class="math inline">\(0.05\)</span>)</li>
</ul></li>
<li>Details:
<ul>
<li>Only genes that pass the threshold for gene expression frequency
will be considered as inputs.</li>
<li>Transform the raw counts to log2 (CPM + 1).</li>
<li>Compute cdr(cellular detection rate).</li>
<li>Run MAST methods on the log2 transformed counts with cellgroups,
repgroups and cdr as covaraites.</li>
<li>MAST fits zero inflated generalized linear models for each gene and
use likelihood ratio test for significance testing. -<a
href="https://rglab.github.io/MAST/articles/MAITAnalysis.html#differential-expression-using-a-hurdle-model">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>log2FC: log2 fold change of counts between the two groups. Positive
values indicate that the gene is more expressed in the first group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if the padj is
smaller than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>
<div id="wilcox" class="section level1">
<h1>Wilcox</h1>
<ul>
<li>Input:
<ul>
<li>object: A Seurat object containing normalized/integrated counts
obtained from Seurat package<br />
</li>
<li>ident.1: Indices of the first group<br />
</li>
<li>ident.2: Indices of the second group</li>
<li>test.use: Denotes which test to use. (“wilcox”:Wilcoxon Rank Sum
test)</li>
<li>min.pct: Only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. (Default is 0.1)</li>
<li>logfc.threshold: Limit testing to genes which show, on average, at
least X-fold difference (log-scale) between the two groups of cells.
(Default is 0.25.)</li>
</ul></li>
<li>Details:
<ul>
<li>Rand-sum test</li>
<li>Run Wilcox methods (Seurat::FindMarkers) on the
normalized/integrated counts.</li>
<li><a
href="https://satijalab.org/seurat/articles/de_vignette">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>avg_log2FC: log2 fold change of mean counts between the two groups.
Positive values indicate that the gene is more expressed in the first
group. <span
class="math inline">\(\log_2(mean_1+1)-\log_2(mean_2+1)\)</span></li>
<li>p_val: Unadjusted p-value</li>
<li>p_val_adj: Adjusted p-value by Bonferroni correction.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if the p_val_adj is
smaller than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>
<div id="mmvst" class="section level1">
<h1>MMvst</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing VST counts provided
by sctransform.</li>
<li>group_id: A feature labeling the cell groups or states</li>
<li>sample_id: A feature labeling the replicates (donors)</li>
</ul></li>
<li>Details:
<ul>
<li>Fitting linear mixed models (∼1+group_id+(1|sample_id)) on VST
data.</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/vignettes/muscat/inst/doc/analysis.html">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>log2FC: log2 fold change of counts between the two groups. Positive
values indicate that the gene is more expressed in the first group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if BH is smaller
than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>
<div id="mmpoisson" class="section level1">
<h1>MMpoisson</h1>
<ul>
<li>Input:
<ul>
<li>sce: A SingleCellExpreriment object containing VST counts provided
by sctransform.</li>
<li>group_id: A feature labeling the cell groups or states</li>
<li>sample_id: A feature labeling the replicates (donors)</li>
</ul></li>
<li>Details:
<ul>
<li>Fitting Poisson generalized linear mixed models
(∼1+group_id+(1|sample_id)) on raw UMI counts.</li>
<li>Library size factors are computed and used as the offset in the
model.<br />
</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/vignettes/muscat/inst/doc/analysis.html">Ref</a></li>
</ul></li>
<li>Output:
<ul>
<li>log2FC: log2 fold change of counts between the two groups. Positive
values indicate that the gene is more expressed in the first group.</li>
<li>pval: Unadjusted p-value</li>
<li>BH: Adjusted p-value by Benjamini-Hochberg procedure.</li>
<li>hits: Indicating whether the gene is a DEG. TRUE if BH is smaller
than 0.05 and absolute log2FC is greater than the predetermined
threshold.</li>
</ul></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
